source/software/singularity.rst

```diff
@@ -0,0 +1,245 @@
+Can I run containers on the HPC systems?
+========================================
+
+The best-known container implementation is doubtlessly `docker`_.  However,
+due to security concerns HPC sites typically don't allow users to run
+Docker containers.
+
+Fortunately, `Singularity`_ addresses container related security issues,
+so Singularity images can be used on the cluster.  Since a Singularity
+image can be built from a docker container, that should not be a severe
+limitation.
+
+
+How can I create a Singularity image?
+-------------------------------------
+
+You have three options to build images, locally on your  machine, in the
+cloud or on the VSC infrastructure.
+
+
+Building on VSC infrastructure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given that most build procedures require superuser privileges, your options
+on the VSC infrastructure are limited.  You can build an image from a docker
+container, e.g., to build an image that contains a version of tensorflow that
+can be used on GPUs, and has jupyter as well, use::
+
+   $ export SINGULARITY_TMPDIR=$VSC_SCRATCH/singularity_tmp
+   $ mkdir -p $SINGULARITY_TMPDIR
+   $ export SINGULARITY_CACHEDIR=$VSC_SCRATCH/singularity_cache
+   $ mkdir -p $SINGULARITY_CACHEDIR
+   $ singularity build tensorflow.sif docker://tensorflow/tensorflow:latest-jupyter
+
+.. warning::
+
+   Don't forget to define and create the ``$SINGULARITY_TMPDIR`` and
+   ``$SINGULARITY_CACHEDIR`` since if you fail to do so, Singularity
+   will use directories in your home directory, and you will exceed
+   the quota on that file system.
+
+   Also, images tend to be very large, so store them on in a directory
+   where you have sufficient quota, e.g., ``$VSC_DATA``.
+
+
+This approach will serve you well when you can use either prebuild images
```

Comment:
typo: prebuilt

Reply:
@smoors Thanks for the feedback.  Meanwhile I've added a 'use cases' section that starts with a remark similar to the one you mention (i.e., use modules whenever you can).

Typos have been fixed.

'container' versus 'image', good question.